src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter.PatternLocalRewriter.cs

```diff
@@ -296,6 +296,9 @@ void addArg(RefKind refKind, BoundExpression expression)
                             return _factory.AssignmentExpression(output, access);
                         }
 
+                    case BoundDagAssignmentEvaluation:
+                        throw ExceptionUtilities.Unreachable;
```

Comment:
🚧 Need to verify this is also unreachable for linear is-expressions (different code path)